x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -66,12 +67,12 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
     return this.runSearch(request, combinedSignal, strategy).pipe(
       expand((response) => {
         // If the response indicates of an error, stop polling and complete the observable
-        if (!response || (!response.isRunning && response.isPartial)) {
+        if (isErrorResponse(response)) {
           return throwError(new AbortError());
```

Comment:
We need to revisit the error we throw here. I think we need to forward the error returned by ES.

Reply:
Yep, this is handled in the error alignment PR

https://github.com/elastic/kibana/pull/77788/files#diff-75f11add5ff5d57a12822fe0c5a52536R91